components/automate-ui/src/app/modules/infra-proxy/treetable/component/treetable.component.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  OnChanges,
+  SimpleChanges
+} from '@angular/core';
+import { Node, TreeTableNode, Options, SearchableNode } from '../models';
+import { TreeService } from '../services/tree/tree.service';
+import { MatTableDataSource } from '@angular/material/table';
+import { ValidatorService } from '../services/validator/validator.service';
+import { ConverterService } from '../services/converter/converter.service';
+import { defaultOptions } from '../default.options';
+import { flatMap, defaults } from 'lodash';
+import { Subject } from 'rxjs';
+import * as O from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+
+@Component({
+  selector: 'app-treetable',
```

Comment:
Could we do this?
```suggestion
  selector: 'app-tree-table',
```
And also change the class name to `TreeTableComponent`.